backend/client.go

```diff
@@ -139,11 +139,24 @@ func (c *backendClient) TerminateOrchestration(ctx context.Context, id api.Insta
 			return fmt.Errorf("failed to configure termination request: %w", err)
 		}
 	}
-
 	e := helpers.NewExecutionTerminatedEvent(req.Output, req.Recursive)
+	// Parent instance needs to be terminated first
 	if err := c.be.AddNewOrchestrationEvent(ctx, id, e); err != nil {
-		return fmt.Errorf("failed to add terminate event: %w", err)
+		return fmt.Errorf("failed to terminate orchestration instance %s: %w", id, err)
 	}
+	if req.Recursive {
+		subOrchestrationInstances, err := GetSubOrchestrationInstances(ctx, c.be, id, false)
```

Comment:
Here although we have added Terminate event to the parent, parent might not actually be terminated before executing this. Can this be an issue? As we are trying to load state and get list of subOrchestrationinstances here even when the Parent possibly might still be running. A possible racy condition?

Reply:
Added terminate handling in backend/orchestration.go